src/hooks/queries.ts

```diff
@@ -54,6 +57,46 @@ export const useGetInfiniteBeefs = (pageSize: number, sort: BeefSortType, search
   })
 }
 
+const useGetAllUserDetailsAction = (addresses?: Address[]) =>
+  useQuery({
+    enabled: addresses !== undefined,
+    queryKey: [queryKeys.getAllUserDetails, addresses],
+    queryFn: () => getAllUserDetailsAction(addresses ?? []),
```

Comment:
OT: Do you have any good resource where I can read about server actions?

Reply:
Hm, Vercel has a write up [here](https://nextjs.org/docs/app/building-your-application/data-fetching/server-actions-and-mutations). 

But I'd say the best way to use them is to treat them exactly like api endpoint handlers but with types. Each server action call is really just a api call that is managed by Next